packages/bindings-factory/lib/Bindings.ts

```diff
@@ -8,12 +12,20 @@ import { bindingsToString } from './bindingsToString';
 export class Bindings implements RDF.Bindings {
   public readonly type = 'bindings';
 
+  public context: IActionContext;
+  private readonly contextMergeHandlers: Record<string, IMergeHandler<any>>;
+
   private readonly dataFactory: RDF.DataFactory;
   private readonly entries: Map<string, RDF.Term>;
 
-  public constructor(dataFactory: RDF.DataFactory, entries: Map<string, RDF.Term>) {
+  public constructor(dataFactory: RDF.DataFactory, entries: Map<string, RDF.Term>,
+    contextMergeHandlers: Record<string, IMergeHandler<any>>,
+    context: IActionContext = new ActionContext()) {
```

Comment:
This is here a major contributor to performance decrease.
Instead of defaulting the context to a newly-constructed object, let's allow it to be undefined.
Then we can cheaply do things like `if(this.context)` later in this file.

Reply:
You're right, I added this to the PR with the implementation of setting, getting, and deleting binding context entries. This does remind me to do a full benchmark to make sure I've not made any mistakes that are not in the merge function. Will get back to you with those results.